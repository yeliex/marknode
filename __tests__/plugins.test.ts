import { Processor } from 'oh-markdown';

const md = `# Hello World {#hello,center}

## list {font-semibold,color=#777777}

{>color=red}
- 1 {color=blue}
- 2

center-test {center}

important {color=red,font-semibold}
multi-line

[link](https://www.github.com) {color=red}

![image](https://github.githubassets.com/assets/mona-loading-default-c3c7aad1282f.gif) {width=100px,center}
        `;


describe('Parse Custom Properties', () => {
    test('should parse custom properties', async () => {
        const processor = new Processor({ debug: true });

        const result = processor.parse(md);

        expect(result).toMatchSnapshot();
    });

    test('should convert custom properties to html', async () => {
        const processor = new Processor({ debug: true });

        const result = await processor.toHtml(md);

        expect(result).toMatchSnapshot();
    });

    test('should convert custom properties to node', async () => {
        const processor = new Processor({ debug: true });

        const result = await processor.toNode(md);

        expect(result).toMatchSnapshot();
    });
});


