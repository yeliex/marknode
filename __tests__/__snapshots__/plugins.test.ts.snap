// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse Custom Properties should convert custom properties to html 1`] = `
{
  "data": [
    {
      "content": "<h1 style="text-align: center" id="hello">Hello World</h1><h2 style="font-weight: 600; color: #777777" id="list">list</h2><ul style="color: red"><li style="color: blue">1</li><li>2</li></ul><p style="text-align: center">center-test</p><p><span style="color: red; font-weight: 600">important</span><br>multi-line</p><p style="color: red"><a href="https://www.github.com">link</a></p><p style="width: 100px; text-align: center"><img src="https://github.githubassets.com/assets/mona-loading-default-c3c7aad1282f.gif" alt="image"></p>",
      "type": "html",
    },
  ],
  "meta": {},
  "title": "Hello World {#hello,center}",
  "toc": undefined,
}
`;

exports[`Parse Custom Properties should convert custom properties to node 1`] = `
{
  "data": [
    {
      "children": [
        {
          "type": "text",
          "value": "Hello World",
        },
      ],
      "properties": {
        "id": "hello",
        "style": {
          "textAlign": "center",
        },
      },
      "tagName": "h1",
      "type": "element",
    },
    {
      "children": [
        {
          "type": "text",
          "value": "list",
        },
      ],
      "properties": {
        "id": "list",
        "style": {
          "color": "#777777",
          "fontWeight": "600",
        },
      },
      "tagName": "h2",
      "type": "element",
    },
    {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "1",
            },
          ],
          "properties": {
            "style": {
              "color": "blue",
            },
          },
          "tagName": "li",
          "type": "element",
        },
        {
          "children": [
            {
              "type": "text",
              "value": "2",
            },
          ],
          "properties": {},
          "tagName": "li",
          "type": "element",
        },
      ],
      "properties": {
        "style": {
          "color": "red",
        },
      },
      "tagName": "ul",
      "type": "element",
    },
    {
      "children": [
        {
          "type": "text",
          "value": "center-test",
        },
      ],
      "properties": {
        "style": {
          "textAlign": "center",
        },
      },
      "tagName": "p",
      "type": "element",
    },
    {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "important",
            },
          ],
          "properties": {
            "style": {
              "color": "red",
              "fontWeight": "600",
            },
          },
          "tagName": "span",
          "type": "element",
        },
        {
          "children": [],
          "properties": {},
          "tagName": "br",
          "type": "element",
        },
        {
          "type": "text",
          "value": "multi-line",
        },
      ],
      "properties": {},
      "tagName": "p",
      "type": "element",
    },
    {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "link",
            },
          ],
          "properties": {
            "href": "https://www.github.com",
          },
          "tagName": "a",
          "type": "element",
        },
      ],
      "properties": {
        "style": {
          "color": "red",
        },
      },
      "tagName": "p",
      "type": "element",
    },
    {
      "children": [
        {
          "children": [],
          "properties": {
            "alt": "image",
            "src": "https://github.githubassets.com/assets/mona-loading-default-c3c7aad1282f.gif",
          },
          "tagName": "img",
          "type": "element",
        },
      ],
      "properties": {
        "style": {
          "textAlign": "center",
          "width": "100px",
        },
      },
      "tagName": "p",
      "type": "element",
    },
  ],
  "meta": {},
  "title": "Hello World {#hello,center}",
  "toc": undefined,
}
`;

exports[`Parse Custom Properties should parse custom properties 1`] = `
{
  "children": [
    {
      "children": [
        {
          "position": {
            "end": {
              "column": 30,
              "line": 1,
              "offset": 29,
            },
            "start": {
              "column": 3,
              "line": 1,
              "offset": 2,
            },
          },
          "type": "text",
          "value": "Hello World {#hello,center}",
        },
      ],
      "depth": 1,
      "position": {
        "end": {
          "column": 30,
          "line": 1,
          "offset": 29,
        },
        "start": {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "heading",
    },
    {
      "children": [
        {
          "position": {
            "end": {
              "column": 38,
              "line": 3,
              "offset": 68,
            },
            "start": {
              "column": 4,
              "line": 3,
              "offset": 34,
            },
          },
          "type": "text",
          "value": "list {font-semibold,color=#777777}",
        },
      ],
      "depth": 2,
      "position": {
        "end": {
          "column": 38,
          "line": 3,
          "offset": 68,
        },
        "start": {
          "column": 1,
          "line": 3,
          "offset": 31,
        },
      },
      "type": "heading",
    },
    {
      "children": [
        {
          "position": {
            "end": {
              "column": 13,
              "line": 5,
              "offset": 82,
            },
            "start": {
              "column": 1,
              "line": 5,
              "offset": 70,
            },
          },
          "type": "text",
          "value": "{>color=red}",
        },
      ],
      "position": {
        "end": {
          "column": 13,
          "line": 5,
          "offset": 82,
        },
        "start": {
          "column": 1,
          "line": 5,
          "offset": 70,
        },
      },
      "type": "paragraph",
    },
    {
      "children": [
        {
          "checked": null,
          "children": [
            {
              "children": [
                {
                  "position": {
                    "end": {
                      "column": 17,
                      "line": 6,
                      "offset": 99,
                    },
                    "start": {
                      "column": 3,
                      "line": 6,
                      "offset": 85,
                    },
                  },
                  "type": "text",
                  "value": "1 {color=blue}",
                },
              ],
              "position": {
                "end": {
                  "column": 17,
                  "line": 6,
                  "offset": 99,
                },
                "start": {
                  "column": 3,
                  "line": 6,
                  "offset": 85,
                },
              },
              "type": "paragraph",
            },
          ],
          "position": {
            "end": {
              "column": 17,
              "line": 6,
              "offset": 99,
            },
            "start": {
              "column": 1,
              "line": 6,
              "offset": 83,
            },
          },
          "spread": false,
          "type": "listItem",
        },
        {
          "checked": null,
          "children": [
            {
              "children": [
                {
                  "position": {
                    "end": {
                      "column": 4,
                      "line": 7,
                      "offset": 103,
                    },
                    "start": {
                      "column": 3,
                      "line": 7,
                      "offset": 102,
                    },
                  },
                  "type": "text",
                  "value": "2",
                },
              ],
              "position": {
                "end": {
                  "column": 4,
                  "line": 7,
                  "offset": 103,
                },
                "start": {
                  "column": 3,
                  "line": 7,
                  "offset": 102,
                },
              },
              "type": "paragraph",
            },
          ],
          "position": {
            "end": {
              "column": 4,
              "line": 7,
              "offset": 103,
            },
            "start": {
              "column": 1,
              "line": 7,
              "offset": 100,
            },
          },
          "spread": false,
          "type": "listItem",
        },
      ],
      "ordered": false,
      "position": {
        "end": {
          "column": 4,
          "line": 7,
          "offset": 103,
        },
        "start": {
          "column": 1,
          "line": 6,
          "offset": 83,
        },
      },
      "spread": false,
      "start": null,
      "type": "list",
    },
    {
      "children": [
        {
          "position": {
            "end": {
              "column": 21,
              "line": 9,
              "offset": 125,
            },
            "start": {
              "column": 1,
              "line": 9,
              "offset": 105,
            },
          },
          "type": "text",
          "value": "center-test {center}",
        },
      ],
      "position": {
        "end": {
          "column": 21,
          "line": 9,
          "offset": 125,
        },
        "start": {
          "column": 1,
          "line": 9,
          "offset": 105,
        },
      },
      "type": "paragraph",
    },
    {
      "children": [
        {
          "position": {
            "end": {
              "column": 11,
              "line": 12,
              "offset": 173,
            },
            "start": {
              "column": 1,
              "line": 11,
              "offset": 127,
            },
          },
          "type": "text",
          "value": "important {color=red,font-semibold}
multi-line",
        },
      ],
      "position": {
        "end": {
          "column": 11,
          "line": 12,
          "offset": 173,
        },
        "start": {
          "column": 1,
          "line": 11,
          "offset": 127,
        },
      },
      "type": "paragraph",
    },
    {
      "children": [
        {
          "children": [
            {
              "position": {
                "end": {
                  "column": 6,
                  "line": 14,
                  "offset": 180,
                },
                "start": {
                  "column": 2,
                  "line": 14,
                  "offset": 176,
                },
              },
              "type": "text",
              "value": "link",
            },
          ],
          "position": {
            "end": {
              "column": 31,
              "line": 14,
              "offset": 205,
            },
            "start": {
              "column": 1,
              "line": 14,
              "offset": 175,
            },
          },
          "title": null,
          "type": "link",
          "url": "https://www.github.com",
        },
        {
          "position": {
            "end": {
              "column": 43,
              "line": 14,
              "offset": 217,
            },
            "start": {
              "column": 31,
              "line": 14,
              "offset": 205,
            },
          },
          "type": "text",
          "value": " {color=red}",
        },
      ],
      "position": {
        "end": {
          "column": 43,
          "line": 14,
          "offset": 217,
        },
        "start": {
          "column": 1,
          "line": 14,
          "offset": 175,
        },
      },
      "type": "paragraph",
    },
    {
      "children": [
        {
          "alt": "image",
          "position": {
            "end": {
              "column": 87,
              "line": 16,
              "offset": 305,
            },
            "start": {
              "column": 1,
              "line": 16,
              "offset": 219,
            },
          },
          "title": null,
          "type": "image",
          "url": "https://github.githubassets.com/assets/mona-loading-default-c3c7aad1282f.gif",
        },
        {
          "position": {
            "end": {
              "column": 108,
              "line": 16,
              "offset": 326,
            },
            "start": {
              "column": 87,
              "line": 16,
              "offset": 305,
            },
          },
          "type": "text",
          "value": " {width=100px,center}",
        },
      ],
      "position": {
        "end": {
          "column": 108,
          "line": 16,
          "offset": 326,
        },
        "start": {
          "column": 1,
          "line": 16,
          "offset": 219,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": {
    "end": {
      "column": 9,
      "line": 17,
      "offset": 335,
    },
    "start": {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
